app/javascript/lib/validation/validate-file-content.js

```diff
@@ -220,6 +236,70 @@ function validateValuesAreNumeric(line, isLastLine, lineNum, dataObj) {
     `Please ensure all values in ${rowText}: ${notedBadRows}, are numbers.`
     issues.push(['error', 'format:invalid-type:not-numeric', msg])
   }
+
+  return issues
+}
+
+/**
+ * Verify sparse matrix column numbers match
+ */
+function validateSparseColumnNumber(line, isLastLine, lineNum, dataObj) {
+  const issues = []
+  dataObj.rowsWithWrongColumnNumbers = dataObj.rowsWithWrongColumnNumbers ? dataObj.rowsWithWrongColumnNumbers : []
+  dataObj.correctNumberOfColumns = dataObj.correctNumberOfColumns ? dataObj.correctNumberOfColumns : ''
+
+  // use the first non-comment or header row to determine correct number of columns
+  if (!line.startsWith('%') && !!line.trim()) {
+    dataObj.correctNumberOfColumns = dataObj.correctNumberOfColumns ? dataObj.correctNumberOfColumns : line.split(' ').length
+    if (dataObj.correctNumberOfColumns !== line.split(' ').length) {
+      dataObj.rowsWithWrongColumnNumbers.push(lineNum)
+    }
+  }
+
+  const numBadRows = dataObj.rowsWithWrongColumnNumbers.length
+  if (isLastLine && numBadRows > 0) {
+    const rowText = numBadRows > 1 ? 'rows' : 'row'
+    const maxLinesToReport = 10
+    let notedBadRows = dataObj.rowsWithWrongColumnNumbers.slice(0, maxLinesToReport).join(', ')
+    if (numBadRows - maxLinesToReport > 0) {
+      notedBadRows += ` and ${numBadRows - maxLinesToReport} more rows`
+    }
+
+    const msg = `All rows must have the same number of columns - ` +
+    `please ensure the number of columns for ${rowText}: ${notedBadRows}, ` +
+    `matches the specificed number of columns-per-row.`
+    issues.push(['error', 'format:mismatch-column-number', msg])
+  }
+
+  return issues
+}
+
+/**
+ * Verify sparse matrix has no blank lines
+ */
+function validateSparseNoBlankLines(line, isLastLine, lineNum, dataObj) {
```

Comment:
wouldn't this check be redundant given that we're also checking the number of columns?

Reply:
It gives a more specific error message that is more helpful I think
<img width="511" alt="Screen Shot 2022-01-18 at 3 30 14 PM" src="https://user-images.githubusercontent.com/54322292/150013905-37bb7bff-bb8a-47ca-8373-bab9117d3afc.png">